### PR TITLE
Support tmux buffer to copy url (with --tmux)

### PR DESCRIPTION
**Pull request by jose1711**
addresses https://github.com/jarun/googler/issues/230

### CODE REVIEW

File: auto-completion/zsh/_googler [hunk 1]
```diff
@@ -58,6 +58,7 @@ args=(
     '(--include-git)--include-git[when used with --upgrade, upgrade to git master]'
     '(- : *)'{-v,--version}'[show version number and exit]'
     '(-d --debug)'{-d,--debug}'[enable debugging]'
+    '(--tmux)--tmux[use tmux for clipboard operations]'
```
Comment (jarun) [inline auto-completion/zsh/_googler]: No new option for this please. PS my next comment.

File: googler [hunk 2]
```diff
@@ -2083,9 +2083,12 @@ class GooglerCmd(object):
                     try:
                         # try copying the url to clipboard using native utilities
                         if sys.platform.startswith(('linux', 'freebsd', 'openbsd')):
+                            if self._opts.tmux and os.getenv('TMUX_PANE'):
+                                copier_params = 'tmux_detected'
+                            else:
```
Comment (jarun) [inline googler]: please change it this way:

```
copier_params = []

if linux-ish et al platform:
    if shutil.which('xsel')
        copier_params = ['xsel', '-b', '-i']
    elif 'TMUX_PANE` available:
        copier_params = ['tmux_detected'] // or append, anything
elif elif sys.platform == 'darwin':
    ...
elif elif sys.platform == 'win32':
    ...
...
```

Comment (jarun) [reply googler]: Check the final code from here as I have updated it.

**Comment by zmwangx** [comment]
Give me a few minutes, I'll do it instead...

**Comment by jarun** [comment]
Sure! If you can test the changes, please go ahead.

**Comment by jarun** [comment]
I probably lost interest in this as soon as I saw tmux in the original issue. it's probably the slowest and most obfuscated piece of crapware I have ever used. I tried it once. Nothing worked out of the box for me. So sorry about my general lack of interest.

**Comment by zmwangx** [comment]
Re tmux: it's not great, but sometimes it's the best you've got. I have 18 panes at the moment. Tmux control mode with a good terminal emulator that supports it (i.e., use your terminal emulator's native windowing, paning, etc.) really take most of the pain out of it.

**Comment by jarun** [comment]
I seldom work remotely nowadays. And multi-tabbed terminator/xfce4-terminal/AltYo in drop-down mode are the poisons for me. ;)

**Comment by jarun** [comment]
Closing this as PR #232 is a better way to handle.

**Comment by jose1711** [comment]
the idea of original PR was that user still has some control of whether the url will be sent to X11 clipboard (when running tmux inside xterm for instance) or passed to tmux. current implementation takes this option away and if tmux is detected it uses tmux'es buffer.

**Comment by zmwangx** [comment]
Unfortunately, we're just averse to adding even more options. Since googler's just a single file and pretty much in feature-freeze at the moment, it shouldn't be hard to maintain a "fork" and have it your way.

**Comment by jose1711** [comment]
yes, you're right. i am thinking.. a compromise could be trying if ```xsel``` won't fail due to undefined ```DISPLAY```. user could then point ```googler``` to X11/tmux by (un)setting ```DISPLAY```. so code-wise: try to execute ```xsel```. if it fails continue detecting tmux pane. what do you think?

**Comment by jarun** [comment]
I believe a much safer (and less complicated) bet is an environment variable `FORCE_TMUX_BUFFER`.

However, given googler's age and popularity and the fact this is the first defect related to `tmux` we are OK with the current status. `xsel` is not even installed on Ubuntu by default (surprise!). If you don't want it, just don't install it or remove it if it is installed in your system. It's too early to put more effort into it.

**Pull request closed without merging**